pkg/sql/opt/xform/custom_funcs.go

```diff
@@ -1388,11 +1388,13 @@ func (c *CustomFuncs) GenerateLookupJoins(
 				constFilters = make(memo.FiltersExpr, 0, numIndexKeyCols-j)
 			}
 
+			foundValType := foundVal.ResolvedType()
```

Comment:
This is probably not right, but no worse than it was before, so I guess we just leave for now...

Once we actually fix the typing, we should add a lint rule to the optimizer tree to not allow calling `ResolvedType`. It's just not safe if we care about OIDs.